### DISCUSSION TITLE
Should an aspiring Smart Contract Developer take the Smart Contract Security/Assembly & Formal Verification Course

### DISCUSSION THREAD

**Discussion by WillStansill** [Cyfrin/foundry-full-course-cu#2143] [Polls]
Hey guys I have finished both the foundry courses and learned an amazing amount of knowledge. I'm confident that Smart Contract Development is the path I want to pursue as a career. With that being said I want to be the best developer I possibly can and wanted advice as to whether or not its recommended for me to invest the time into the three security courses I have yet to get to. Or if my time would be better spent building projects on my own time, adding to my resume, participating in hackathons, etc. Any and all advice and experience would be incredibly appreciated and I look forward to hearing your thoughts.

**Comment by EngrPips** [comment]
Hello @WillStansill, I am convinced that in the web3 world, smart contract security is equally important as smart contract development. No one will be a complete web3 developer if they don't have sound knowledge of smart contract security. So you should do the security course as it is equally important as the foundry-full course, so you will be writing fewer buggy codes.
